db/model/aspect.js

```diff
@@ -192,19 +193,30 @@ module.exports = function aspect(seq, dataTypes) {
       }, // hooks.afterCreate
 
       /**
-       * Set the isDeleted timestamp.
+       * Check Generator references and set the isDeleted timestamp.
        *
        * @param {Aspect} inst - The instance being destroyed
        * @returns {Promise}
        */
       beforeDestroy(inst /* , opts */) {
-        return new seq.Promise((resolve, reject) =>
-          common.setIsDeleted(seq.Promise, inst)
-          .then(() => resolve(inst))
-          .catch((err) => reject(err))
-        );
+        return inst.checkGeneratorReferences('delete')
```

Comment:
Ideally, something like this should be a db association between the model and the generator model like the ones between L141 and L154.  Considering that the use case is slightly different does it make sense to create an association on the db level?

Reply:
I don't see any reason to NOT make it an association... I can't remember why we didn't make it one in the first place.